### DISCUSSION TITLE
Problem with shelly RGBW2 external input

### DISCUSSION THREAD

**Discussion by BartoszGan** [RavenSystem/esp-homekit-devices#1893] [Q&A] [answered]
I have a problem with the Shelly RGBW2. From the appalication everything works fine but sometimes the external momentary switch does not work. Unfortunately I have not found a correlation when it works. The switch is connected to AC and the configuration is as follows:

`{"c":{"l":2,"b":[{"g":13,"t":5}]},"a":[{"t":30,"g":[12,15,14,4],"ld":[0,1,0,1],"b":[{"g":5,"p":0,"i":1,"t":0}]}]}`

Maybe someone has an idea what could be wrong?

**Comment by RavenSystem** [comment, marked as answer]
Try enabling pulse mode:

`{"c":{"l":2,"b":[{"g":13,"t":5}]},"a":[{"t":30,"g":[12,15,14,4],"ld":[0,1,0,1],"b":[{"g":5,"p":0,"i":1,"t":0,"m":1}]}]}`

**Reply by BartoszGan** [reply]
Thank you very much for your answer, now everything works correctly.
